worlds/sc2/options.py

```diff
@@ -806,7 +806,7 @@ class ExcludedMissions(Sc2MissionSet):
     Doesn't apply to vanilla mission order.
     It may be impossible to build a valid campaign if too many missions are excluded."""
     display_name = "Excluded Missions"
-    valid_keys = {mission.mission_name for mission in SC2Mission}
+    valid_keys = lookup_name_to_mission.keys()
```

Comment:
This change isn't necessary; it only validates against mission keys after expanding the groups, and after expansion only literal mission names are valid. See Sc2MissionSet earlier in the file for where expansion happens.

Reply:
Okay, but when I revert that change it throws an error when trying to exclude Zero Hour.

Reply:
You don't need to mark everything down as a literal. The validation calls `Sc2MissionSet.verify()`, which looks at the mission groups to convert all group names into their constituent mission names, if something is not a group or missino name it leaves it in-place, and _then_ it validates that list against the `valid_keys`.

So if you put in "Zero Hour", it will expand that to ["Zero Hour (Terran)", ...], then check that all those exist in the valid_keys, which they do. If you put in "asdf", it will not expand it as it's not a group name, then error out as "asdf" isn't a valid mission name.

Essentially, by going through mission_groups, all the validation and expansion is already handled for you, you don't have to change anything in options.

Reply:
The reason I ask is that when I revert that change to options.py, test_generation fails because "Zero Hour" is not considered valid.

Reply:
Sounds like something may be wrong with where you're defining the mission groups, then, as it would fundamentally be testing the same thing as `test_excluding_mission_groups_excludes_all_missions_in_group()`

Reply:
Just copy-pasted your loop from mission_groups.py and your `test_excluding_mission_short_name_excludes_all_variants_of_mission()` test, ran all tests, and all passed.